### PR TITLE
add directory filter

### PR DESCRIPTION
**Pull request by herlon214**
* Created a `Filter` struct with a function to filter directory only.
* Added a `directory-only` flag which is `false` by default.

Possible fixes https://github.com/nikolaydubina/go-cover-treemap/issues/6

Example how hugo looks:
out

### CODE REVIEW

**Comment by nikolaydubina** [comment]
Yes, you are right parent should be weighted average of children. Let me check...

File: covertreemap/filter.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+func (f Filter) FilterTree(tree *treemap.Tree) {
+	// Check if directory only filter is set
+	if f.DirectoryOnly {
+		f.filterDirectoriesOnly(tree)
+	}
+}
+
+// filterDirectoriesOnly removes the node leafs and keep only the
+// parent directories
+func (f Filter) filterDirectoriesOnly(tree *treemap.Tree) {
```
[the remark this answers is not in the document]
Comment (nikolaydubina) [reply covertreemap/filter.go:24]: Or even can make a single function if you like :)

I checked, most of code in /treemap module is using simple mutator like functions of signature `func(tree *treemap.Tree)` so function name or interface does not matter as much. So good to keep simple. 
```go
func RemoveFilesTreeFilter(tree *treemap.Tree) {
```

**Comment by nikolaydubina** [comment]
No rush tho! I will integrate this later and add you to contributors (Readme) anyways in coming weeks! Or if you want to be in "GitHub Contributors", ok too, just need to merge this PR then.

File: covertreemap/filter.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	newNodes := make(map[string]treemap.Node, 0)
+	newTo := make(map[string][]string, 0)
```
[the remark this answers is not in the document]
Comment (nikolaydubina) [reply covertreemap/filter.go:26]: no need, since imputer already filled all parents hierarchy.

**Comment by herlon214** [comment]
@nikolaydubina I'm checking the output, the coverage value isn't being imputed in the parent (looks like):

```
github.com/gohugoio/hugo/tpl/templates {github.com/gohugoio/hugo/tpl/templates templates 10 0 false}
github.com/gohugoio/hugo/releaser {github.com/gohugoio/hugo/releaser releaser 353 0 false}
github.com/gohugoio/hugo/tpl/path {github.com/gohugoio/hugo/tpl/path path 53 0 false}
github.com/gohugoio/hugo/tpl/images {github.com/gohugoio/hugo/tpl/images images 34 0 false}
github.com/gohugoio/hugo/cache {github.com/gohugoio/hugo/cache cache 270 0 false}
```

**Comment by herlon214** [comment]
Now it makes sense, only the size is being imputed:
```
	sizeImputer := treemap.SumSizeImputer{EmptyLeafSize: 1}
	sizeImputer.ImputeSize(*tree)
```

**Comment by herlon214** [comment]
Looks like adding the weight imputer did the trick:
```
	weightImputer := treemap.WeightedHeatImputer{EmptyLeafHeat: 1}
	weightImputer.ImputeHeat(*tree)
```

**Comment by nikolaydubina** [comment]
you are right, need to impute. it is done a bit later in main.
 
image

ok, looks almost good! thanks for contribution! I will merge this now and polish submission a bit!

as for math.Log or something more special.. don't think we should add very custom logic to main here. The idea is that if you have something very custom like math.Log or data wrangling, then can define main.go in new project and copy main from this file + use this packages to do basic processing:
```
	"github.com/nikolaydubina/go-cover-treemap/covertreemap"
	"github.com/nikolaydubina/treemap"
	"github.com/nikolaydubina/treemap/render"
```

Feel free to open separate submission PR / Issue if you found interesting data step visualisation / use-case if you think that has to be in default case for everyone!

Cheers!

**Comment by nikolaydubina** [comment]
Integrated https://github.com/nikolaydubina/go-cover-treemap#appendix-d-only-folders

🤝

**Pull request merged**